### ISSUE TITLE
javascript:void(0) in <a> is breaking the placeholder

### ISSUE THREAD

**Issue by skoub** [mathiasbynens/jquery-placeholder#74]
When you click on a link that use javascript:void(0) as the href, the placeholder disapear in IE7 & IE8 (maybe others too)

```html
<a href="javascript:void(0)" onClick="doSomething()">
```

**Comment by antoinebrault** [comment]
Maybe something along the lines :

```javascript
var last_clicked;
$("body").on('click', 'a', function() {
	last_clicked = this;
    return true;
});
		
// Clear placeholder values upon page reload
$(window).bind('beforeunload.placeholder', function(e) {
	if (last_clicked && last_clicked.href !== 'javascript:void(0)') {
		$('.'+settings.customClass).each(function() {
			this.value = '';
		});
	}
});
```

might work...

**Comment by rRudeBoy** [comment]
I can confirm this behavior is still occurring in IE9, Aug 2015.  Regarding the link Sangdol posted giving a positive reason for using javascript:void(0), deeper in that answer it's suggested that best practice is to simply not have an href attribute, which I found to be a workable solution.

**Comment by RASMiranda** [comment]
Still happening on IE9.
Can confirm that @paranoida simple workaround prevents de behavior.

**Comment by amerikan** [comment]
Using `<a href="javascript:void(0);" >Click</a>` in IE will trigger a `beforeunload` event. Currently the plugin will clear input values for beforeunload events. So I think @antoinebrault is in the right track.

**Comment by amerikan** [comment]
I've fixed this in this commit: https://github.com/mathiasbynens/jquery-placeholder/commit/891fa637a8b27d00b394f842a596a47beb12f335 

I tested it myself and with this, it is fix. Can any of you test the fix to confirm it?

**Comment by RASMiranda** [comment]
Not working, some behavior happening with v2.2.0

**Comment by amerikan** [comment]
@RASMiranda the fix is not in v2.2.0 yet. Try including this https://github.com/mathiasbynens/jquery-placeholder/blob/gh-pages/jquery.placeholder.js in your script. I'm waiting to fully confirm before release the fix in v2.2.1

**Comment by RASMiranda** [comment]
@amerikan Ha didn't realised it was not realeased, tried with https://github.com/mathiasbynens/jquery-placeholder/blob/gh-pages/jquery.placeholder.js same behavior :-/
But maybe my case is not the best. It has some other plugins that might be messing up and some other stuff that I don't know off.
I'll have to live with the workaround for now, don't have much time to test it further.
